### PR TITLE
Adding libfilepattern recipe

### PR DESCRIPTION
**Pull request by sameeul**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sameeul** [comment]
@JesseMckinzie: Can you please acknowledge here that you will be  a recipe maintainer for this package?

**Comment by JesseMckinzie** [comment]
I will be a recipe maintainer for this package

**Comment by sameeul** [comment]
https://github.com/orgs/conda-forge/teams/help-c-cpp : Ready for review.

File: recipes/libfilepattern/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+set -eux # Abort on error.
+
+mkdir build
+cd build
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+	export CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+fi
+
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=${PREFIX} -DCMAKE_INSTALL_PREFIX=${PREFIX} -Dfilepattern_SHARED_LIB=ON  ../src/filepattern/cpp/
```
Comment (xhochy) [inline recipes/libfilepattern/build.sh:12]:
```suggestion
cmake $CMAKE_ARGS -DCMAKE_BUILD_TYPE=Release -Dfilepattern_SHARED_LIB=ON  ../src/filepattern/cpp/
```

File: recipes/libfilepattern/bld.bat [hunk 2]
```diff
@@ -0,0 +1,10 @@
+mkdir build
+cd build
+
+
+cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_CXX_FLAGS_RELEASE="/MT /O2 /D NDEBUG" -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ../src/filepattern/cpp/
```
Comment (xhochy) [inline recipes/libfilepattern/bld.bat:5]:
```suggestion
cmake %CMAKE_ARGS% -DBUILD_SHARED_LIBS=ON -DCMAKE_CXX_FLAGS_RELEASE="/MT /O2 /D NDEBUG" ../src/filepattern/cpp/
```

File: recipes/libfilepattern/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+cmake --build . --target install --parallel
```
Comment (xhochy) [inline recipes/libfilepattern/build.sh:14]:
```suggestion
cmake --build . --target install --parallel -j${CPU_COUNT}
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by sameeul** [comment]
Thanks for looking at it. I updated the code as suggested.

**Comment by sameeul** [comment]
@xhochy: I have updated the code as suggested. Is it good to merge now?

**Review by xhochy — APPROVED** [review]

**Pull request merged**